joint_state_publisher/joint_state_publisher/joint_state_publisher.py

```diff
@@ -90,59 +90,66 @@ def init_urdf(self, robot):
                         maxval = float(limit.getAttribute('upper'))
-                    except:
-                        self.get_logger().warn('%s is not fixed, nor continuous, but limits are not specified!' % name)
-                        continue
-
-                safety_tags = child.getElementsByTagName('safety_controller')
-                if self.use_small and len(safety_tags) == 1:
-                    tag = safety_tags[0]
-                    if tag.hasAttribute('soft_lower_limit'):
-                        minval = max(minval, float(tag.getAttribute('soft_lower_limit')))
-                    if tag.hasAttribute('soft_upper_limit'):
-                        maxval = min(maxval, float(tag.getAttribute('soft_upper_limit')))
-
-                mimic_tags = child.getElementsByTagName('mimic')
-                if self.use_mimic and len(mimic_tags) == 1:
-                    tag = mimic_tags[0]
-                    entry = {'parent': tag.getAttribute('joint')}
-                    if tag.hasAttribute('multiplier'):
-                        entry['factor'] = float(tag.getAttribute('multiplier'))
-                    if tag.hasAttribute('offset'):
-                        entry['offset'] = float(tag.getAttribute('offset'))
-
-                    self.dependent_joints[name] = entry
+                except ValueError:
+                    self.get_logger().warn('%s limits are not valid!' % name)
                     continue
-
-                if name in self.dependent_joints:
+                except:
+                    self.get_logger().warn('%s is not fixed, nor continuous, but limits are not specified!' % name)
                     continue
 
-                if self.zeros and name in self.zeros:
-                    zeroval = self.zeros[name]
-                elif minval > 0 or maxval < 0:
-                    zeroval = (maxval + minval)/2
-                else:
-                    zeroval = 0
+            safety_tags = child.getElementsByTagName('safety_controller')
+            if self.use_small and len(safety_tags) == 1:
+                tag = safety_tags[0]
+                if tag.hasAttribute('soft_lower_limit'):
+                    minval = max(minval, float(tag.getAttribute('soft_lower_limit')))
+                if tag.hasAttribute('soft_upper_limit'):
+                    maxval = min(maxval, float(tag.getAttribute('soft_upper_limit')))
+
+            mimic_tags = child.getElementsByTagName('mimic')
+            if self.use_mimic and len(mimic_tags) == 1:
+                tag = mimic_tags[0]
+                entry = {'parent': tag.getAttribute('joint')}
+                if tag.hasAttribute('multiplier'):
+                    entry['factor'] = float(tag.getAttribute('multiplier'))
+                if tag.hasAttribute('offset'):
+                    entry['offset'] = float(tag.getAttribute('offset'))
+
+                self.dependent_joints[name] = entry
+                continue
+
+            self.joint_list.append(name)
```

Comment:
Ah, sorry.  I made a mistake here.  This needs to go above the `mimic_tags` above, since the `mimic_tags` could conceivably change the state of `self.dependent_joints`.